samples/java-spring-shopping-cart-quickstart/src/main/java/shoppingcart/domain/ShoppingCart.java

```diff
@@ -0,0 +1,69 @@
+package shoppingcart.domain;
+
+import kalix.javasdk.annotations.TypeName;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public record ShoppingCart(String cartId, List<LineItem> items, boolean checkedOut) {
+
+
+  public record LineItem(String productId, String name, int quantity) {
+    public LineItem increaseQuantity(int qty) {
+      return new LineItem(productId, name, quantity + qty);
+    }
+  }
+
+  public ShoppingCart addItem(LineItem item) {
+    var itemToAdd =
+      items.stream()
```

Comment:
A lot of boilerplate code to update sth in the list, with a map it will be much simpler, but then a serialised version will look worse.

Reply:
Agree. I also considered using a Map, but decide to leave it as is. Note that similar code is also added to the event sourced sample.